packages/ic-response-verification/src/verification/verify_request_response_pair.rs

```diff
@@ -272,14 +273,23 @@ fn v2_verification(
     );
 
     match are_hashes_valid {
-        true => Ok(VerificationInfo {
-            response: Some(VerifiedResponse {
-                status_code: Some(response.status_code),
-                headers: response_headers.headers,
-                body: response.body,
-            }),
-            verification_version: 2,
-        }),
+        true => {
+            let certificate_headers: Vec<(String, String)> = vec![(
```

Comment:
nit: the variable should probably be named `certificate_header` (i.e. singular), as there is only one header.
Also, one could skip creation of the vector with one element, and just use `all_headers.push(...)`